test/spi/src/main/java/org/jboss/arquillian/test/spi/execution/TestExecutionDecider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.arquillian.test.spi.execution;
+
+import java.lang.reflect.Method;
+
+/**
+ * @author <a href="mailto:smikloso@redhat.com">Stefan Miklosovic</a>
+ * 
+ */
+public interface TestExecutionDecider
+{
+
+    static final TestExecutionDecider EXECUTE = new TestExecutionDecider()
+    {
+
+        @Override
+        public boolean execute(Method testMethod)
+        {
+            return true;
+        }
+
+    };
+
+    static final TestExecutionDecider DONT_EXECUTE = new TestExecutionDecider()
+    {
+
+        @Override
+        public boolean execute(Method testMethod)
+        {
+            return false;
+        }
+
+    };
+
+    /**
+     * 
+     * @param testMethod test method to resolve a test execution on
+     * @return true if execution is performed, false otherwise
+     */
+    boolean execute(Method testMethod);
```

Comment:
Wonder if more then just a bool would be apprichiated here. .e.g. a Cause Message for false. 

Somewhere we might want to say something about; who stopped it from executing and why.